### ISSUE TITLE
Cannot build a project with space in the name

### ISSUE THREAD

**Issue by blowdart** [dotnet/cli#3221]
## Steps to reproduce

Create an xproj with a space in the file name or in the containing directory. I encountered this trying to port my authorization lab from https://github.com/blowdart/AspNetAuthorizationWorkshop. As you can see there are multiple Step X - Description projects. 
## Expected  behavior

Compilation works.
## Actual behavior

When running dotnet build 

```
D:\PersonalCode\AspNetAuthorizationWorkshop\src\Step 1 - Setup Authorization>dotnet build
Project Step 1 - Setup Authorization (.NETCoreApp,Version=v1.0) will be compiled because expected outputs are missing
Compiling Step 1 - Setup Authorization for .NETCoreApp,Version=v1.0
C:\Program Files\dotnet\dotnet.exe compile-csc @D:\PersonalCode\AspNetAuthorizationWorkshop\src\Step 1 - Setup Authorization\obj\Debug\netcoreapp1.0\dotnet-compile.rsp returned Exit Code 1
D:\PersonalCode\AspNetAuthorizationWorkshop\src\Step 1 - Setup Authorization\error CS2007: Unrecognized option: '-'

Compilation failed.
    0 Warning(s)
    1 Error(s)

```
## Environment data

:\PersonalCode\AspNetAuthorizationWorkshop\src\Step 1 - Setup Authorization>dotnet --info
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

D:\PersonalCode\AspNetAuthorizationWorkshop\src\Step 1 - Setup Authorization>

**Comment by blackdwarf** [comment]
@blowdart this is xproj specific?

**Comment by blowdart** [comment]
Well, that's all I use, so I would assume so :) They're all ASP.NET Core apps.

**Comment by phenning** [comment]
I did some initial investigation as we discovered this was happening with Web templates but not console templates. 

It looks like an issue with setting "preserveCompilationContext": true.

Repo steps are: 

1. mkdir "c:\dd\path with spaces"
2. dotnet new
3. add "preserveCompilationContext": true to the buildOptions in project json
4. dotnet restore
5. dotnet build

results in:
C:\dd\path with space\error CS2001: Source file 'C:\dd\path with space\space.deps.json' could not be found.
C:\dd\path with space\error CS2001: Source file 'C:\dd\path with space\with' could not be found.

obj.\debug\netcoreapp1.0\dotnet-compile-csc.rsp contains:
-resource:"C:\dd\path with space\obj\Debug\netcoreapp1.0\path with spacedotnet-compile.deps.json",path with space.deps.json

[dotnet-compile-csc.rsp.txt](https://github.com/dotnet/cli/files/284866/dotnet-compile-csc.rsp.txt)
